code/modules/mob/mob_movement.dm

```diff
@@ -15,21 +15,27 @@
 			return 1
 	return (!mover.density || !density || lying)
 
+//The byond version of these verbs wait for the next tick before acting.
+//	instant verbs however can run mid tick or even during the time between ticks.
+/client/verb/moveup()
+	set name = ".moveup"
+	set instant = 1
+	Move(get_step(mob, NORTH), NORTH)

```

Comment:
I don't know if you bother with client view rotation at all, but if you do and wish for key-presses to remain consistent (i.e. pressing the Up-key should move you up on the screen, not the north direction of the map) you may want to look at https://github.com/Baystation12/Baystation12/pull/16973

However, I can also see a case for keeping the current implementation as it means you won't smack into a wall if you run down a corridor and the view suddenly rotates 90 degrees (or turn around in case of a 180 degree rotation).